components/org.wso2.carbon.identity.application.authenticator.oidc/src/test/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticatorTest.java

```diff
@@ -92,166 +142,296 @@
-        Assert.assertNotNull(openIDConnectAuthenticator.getContextIdentifier(mockServletRequest), "Invalid context identifier.");
+    @Test(dataProvider = "logEnabler")
+    public void testPassProcessAuthenticationWithParamValue(boolean debugEnabled) throws Exception {
+        setupTest();
+        setMockedLog(debugEnabled);
+        authenticatorProperties.put("callbackUrl", "http://localhost:8080/playground2/oauth2client");
+        Map<String, String> paramMap = new HashMap<>();
+        paramMap.put("redirect_uri","http:/localhost:9443/oauth2/redirect");
+        when(mockAuthenticationContext.getProperty("oidc:param.map")).thenReturn(paramMap);
+        setParametersForOAuthClientResponse(mockOAuthClientResponse, accessToken, idToken);
+        openIDConnectAuthenticator.processAuthenticationResponse(mockServletRequest,
+                mockServletResponse, mockAuthenticationContext);
+    }
 
-        when(mockServletRequest.getParameter(anyString())).thenReturn(null);
-        Assert.assertNull(openIDConnectAuthenticator.getContextIdentifier(mockServletRequest), "Invalid context identifier.");
+    @Test(dataProvider = "seperator")
+    public void testBuildClaimMappings(String separator) throws Exception {
+        Map<ClaimMapping, String> claims = new HashMap<>();
+        Map<String, Object> entries = new HashMap<>();
+        entries.put("scope", new Object());
+
+        for (Map.Entry<String, Object> entry : entries.entrySet()) {
+            openIDConnectAuthenticator.buildClaimMappings(claims, entry, separator);
+            Assert.assertTrue(!claims.isEmpty(), "Claims[] is empty.");
+        }
+        entries = new HashMap<>();
+        entries.put("scope", new String("[    \n" +
+                "    {\"name\":\"Ram\", \"email\":\"ram@gmail.com\", \"age\":23},    \n" +
+                "    {\"name\":\"Shyam\", \"email\":\"shyam23@gmail.com\", \"age\":28},  \n" +
+                "]"));
+        for (Map.Entry<String, Object> entry : entries.entrySet()) {
+            setMockedLog(true);
+            openIDConnectAuthenticator.buildClaimMappings(claims, entry, separator);
+            Assert.assertTrue(!claims.isEmpty(), "Claims[] is empty.");
+        }
     }
 
-    @Test(priority = 1)
+    @Test(dataProvider = "requestDataHandler")
+    public void testGetContextIdentifier(String grantType, String state, String loginType, String error, String debugEnable) throws Exception {
+        setMockedLog(Boolean.parseBoolean(debugEnable));
+        when(mockServletRequest.getParameter(OIDCAuthenticatorConstants.OAUTH2_GRANT_TYPE_CODE)).thenReturn(grantType);
+        when(mockServletRequest.getParameter(OIDCAuthenticatorConstants.OAUTH2_PARAM_STATE)).thenReturn(state);
+        when(mockServletRequest.getParameter(OIDCAuthenticatorConstants.LOGIN_TYPE)).thenReturn(loginType);
+        when(mockServletRequest.getParameter(OIDCAuthenticatorConstants.OAUTH2_ERROR)).thenReturn(error);
+
+        if (error != null) {
```

Comment:
you can get the expected result from DataProvider and use `Assert.assertEquals(openIDConnectAuthenticator.getContextIdentifier(mockServletRequest), expected)`